src/legacy/views/secondary-bar/parts/edit/share-folder-properties.tsx

```diff
@@ -3,55 +3,73 @@
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
+import React, { FC, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+
 import {
 	ButtonOld as Button,
 	Chip,
+	ChipProps,
 	Container,
 	Padding,
+	SnackbarManagerContext,
 	Text,
 	Tooltip
 } from '@zextras/carbonio-design-system';
-import { useUserAccounts } from '@zextras/carbonio-shell-ui';
-import { map, replace, split } from 'lodash';
-import React, { useCallback, useContext, useMemo, useState } from 'react';
+import { Grant, soapFetch, useUserAccounts } from '@zextras/carbonio-shell-ui';
+import { map } from 'lodash';
 import { useTranslation } from 'react-i18next';
 import styled from 'styled-components';
-import { sendShareNotification } from '../../../../store/actions/send-share-notification';
-import { ShareFolderRoleOptions, findLabel } from '../../commons/utils';
-import { capitalise } from '../../utils';
+
 import { Context } from './edit-context';
 import { useAppDispatch } from '../../../../hooks/redux';
+import { sendShareNotification } from '../../../../store/actions/send-share-notification';
+import {
+	ActionProps,
+	GranteeInfoProps,
+	GranteeProps,
+	ShareFolderPropertiesProps
+} from '../../../../types/contact';
+import { ShareFolderRoleOptions, findLabel } from '../../commons/utils';
 
-const HoverChip = styled(Chip)`
-	background-color: ${({ theme, hovered }) =>
+const HoverChip = styled(Chip)<ChipProps & { hovered?: boolean }>`
+	background-color: ${({ theme, hovered }): string =>
 		hovered ? theme.palette.gray3.hover : theme.palette.gray3.regular};
 `;
-export const GranteeInfo = ({ grant, shareFolderRoleOptions, hovered }) => {
+
+export const GranteeInfo: FC<GranteeInfoProps> = ({ grant, shareFolderRoleOptions, hovered }) => {
 	const role = useMemo(
 		() => findLabel(shareFolderRoleOptions, grant.perm || ''),
 		[shareFolderRoleOptions, grant.perm]
 	);
-	const label = useMemo(
-		() => `${replace(split(grant.d, '@')?.[0], '.', ' ')} - ${role}`,
-		[grant.d, role]
-	);
-	const upperCaseLabel = useMemo(() => capitalise(label), [label]);
+
+	const label = useMemo(() => {
+		const composeLabel = (name: string): string => `${name} - ${role}`;
+		return grant.d ? composeLabel(grant.d) : composeLabel(grant?.zid);
+	}, [grant, role]);
+
 	return (
 		<Container crossAlignment="flex-start">
 			<Text>
-				<HoverChip label={upperCaseLabel} hovered={hovered} />
+				<HoverChip label={label} hovered={hovered} />
 			</Text>
 		</Container>
 	);
 };
 
-const Actions = ({ folder, grant, createSnackbar, setActiveModal, onMouseLeave, onMouseEnter }) => {
+const Actions: FC<ActionProps> = ({
+	folder,
+	grant,
+	setActiveModal,
+	onMouseLeave,
+	onMouseEnter
+}) => {
 	const [t] = useTranslation();
 	const accounts = useUserAccounts();
+	const createSnackbar = useContext(SnackbarManagerContext);
```

Comment:
Use `useSnackbar` hook instead of the context. The specific hook always return a valued function.
```suggestion
	const createSnackbar = useSnackbar();
```